### PR TITLE
DEV: Show message when cannot invite user to PM

### PR DESCRIPTION
**Pull request by oblakeerickson**
When inviting a user to a PM return a message that says, "Sorry, this
user can't be invited." if they have been muted or are not in a users
allowed pm users list.

### CODE REVIEW

File: app/controllers/topics_controller.rb [hunk 1]
```diff
@@ -691,6 +691,8 @@ def invite
       end
     rescue Topic::UserExists => e
```
Comment (eviltrout) [inline app/controllers/topics_controller.rb:692]: Both of these exceptions return the same thing so you could handle both on one line with a comma.

File: app/models/topic.rb [hunk 2]
```diff
@@ -1039,6 +1044,43 @@ def invite_existing_muted?(target_user, invited_by)
     false
   end
 
+  def allowed_pm_user?(target_user, invited_by)
+    return true if target_user.staff?
+    users = TopicAllowedUser.where(topic: self).pluck(:user_id)
```
Comment (eviltrout) [inline app/models/topic.rb:1049]: Could we also return early if `invited_by.staff?`

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -244,6 +244,7 @@ en:
   topic_invite:
     failed_to_invite: "The user cannot be invited into this topic without a group membership in either one of the following groups: %{group_names}."
     user_exists: "Sorry, that user has already been invited. You may only invite a user to a topic once."
+    not_allowed: "Sorry, this user can't be invited."
```
Comment (eviltrout) [inline config/locales/server.en.yml:247]:
```suggestion
    not_allowed: "Sorry, that user can't be invited."
```
(consistent with the translation above it.)

File: config/locales/client.en.yml [hunk 4]
```diff
@@ -2465,6 +2465,7 @@ en:
         success: "We've invited that user to participate in this message."
         success_group: "We've invited that group to participate in this message."
         error: "Sorry, there was an error inviting that user."
+        not_allowed: "Sorry, this user can't be invited."
```
Comment (eviltrout) [inline config/locales/client.en.yml:2468]:
```suggestion
        not_allowed: "Sorry, that user can't be invited."
```
(consistent with other translations above)

File: app/models/topic.rb (hunk 2 above)
Comment (oblakeerickson) [reply app/models/topic.rb:1049]: Actually I think not? Here is a scenario: A conversation between StaffA and UserB has started. UserB has allowed_pm_users enabled and only has allowed StaffA and UserD. StaffA invites UserC, but UserB hasn't allowed them so they can't be added to the pm.

Comment (eviltrout) [reply app/models/topic.rb:1049]: That is an interesting edge case. I kinda feel like staff should be able to say "do this anyway" but I also see your point of view. Let's keep it your way and see how it goes.

**Pull request merged**